clustered/integration-test/src/test/java/org/ehcache/clustered/ReconnectDuringDestroyTest.java

```diff
@@ -0,0 +1,121 @@
+import org.terracotta.testing.rules.Cluster;
+
+import com.tc.net.proxy.TCPProxy;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.ehcache.clustered.reconnect.BasicCacheReconnectTest.RESOURCE_CONFIG;
+import static org.ehcache.clustered.util.TCPProxyUtil.setDelay;
+import static org.terracotta.testing.rules.BasicExternalClusterBuilder.newCluster;
+
+/**
+ * ReconnectDuringDestroyTest
+ */
+public class ReconnectDuringDestroyTest extends ClusteredTests {
+  private static PersistentCacheManager cacheManager;
+  private static ServerSideConfigurationBuilder serverSideConfigurationBuilder;
+  private static final List<TCPProxy> proxies = new ArrayList<>();
+
+  @ClassRule
+  public static Cluster CLUSTER =
+    newCluster().in(new File("build/cluster")).withServiceFragment(RESOURCE_CONFIG).build();
+
+  @BeforeClass
+  public static void waitForActive() throws Exception {
+    CLUSTER.getClusterControl().waitForActive();
+
+    URI connectionURI = TCPProxyUtil.getProxyURI(CLUSTER.getConnectionURI(), proxies);
+
+    serverSideConfigurationBuilder = ClusteringServiceConfigurationBuilder.cluster(connectionURI
+      .resolve("/crud-cm"))
+      .autoCreate()
+      .defaultServerResource("primary-server-resource");
+
+    CacheManagerBuilder<PersistentCacheManager> clusteredCacheManagerBuilder
+      = CacheManagerBuilder.newCacheManagerBuilder()
+      .with(serverSideConfigurationBuilder);
+    cacheManager = clusteredCacheManagerBuilder.build(false);
+    cacheManager.init();
+  }
+
+  @Test
+  public void reconnectDuringDestroyTest() throws Exception {
+    Connection client = CLUSTER.newConnection();
+    VoltronReadWriteLock voltronReadWriteLock = new VoltronReadWriteLock(client, "crud-cm");
+    try (VoltronReadWriteLock.Hold localMaintenance = voltronReadWriteLock.tryWriteLock()) {
+      if (localMaintenance == null) {
+        throw new EntityBusyException("Unable to obtain maintenance lease for ");
+      }
+      EntityRef<ClusterTierManagerClientEntity, ClusterTierManagerConfiguration, Void> ref = getEntityRef(client);
+      try {
+        ClusterTierManagerClientEntity entity = ref.fetchEntity(null);
+        entity.prepareForDestroy();
+        entity.close();
+      } catch (EntityNotFoundException e) {
+        Assert.fail();
+      }
+      setDelay(6000, proxies);
+      Thread.sleep(6000);
+      setDelay(0L, proxies);
+    }
+    voltronReadWriteLock = new VoltronReadWriteLock(client, "crud-cm");
+    try (VoltronReadWriteLock.Hold localMaintenance = voltronReadWriteLock.tryWriteLock()) {
+      if (localMaintenance == null) {
+        throw new EntityBusyException("Unable to obtain maintenance lease for ");
+      }
+      EntityRef<ClusterTierManagerClientEntity, ClusterTierManagerConfiguration, Void> ref = getEntityRef(client);
+      try {
+        ClusterTierManagerClientEntity entity = ref.fetchEntity(null);
+        entity.prepareForDestroy();
```

Comment:
prepare is being done before introducing the delay, right? Why do it here again?

Reply:
Because in reconnect path it tries to again repeat the same set of calls for cacheManager.destroy().